feat: add `FIF_F` macro in `math/base/napi/ternary`

Neerajpathak07 opened the pull request:
Resolves none.

## Description

> What is the purpose of this pull request?

This pull request:

-   adds `FIF_F` macro in `math/base/napi/ternary`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves none.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-  [x] Read, understood, and followed the [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md)

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

Neerajpathak07 committed:
feat: adding essential files

Neerajpathak07 committed:
chore: cleaning markdown

anandkaranubc commented on lib/node_modules/@stdlib/math/base/napi/ternary/src/fif_f.c:
```diff
@@ -0,0 +1,98 @@
+#include <assert.h>
+
+/**
+* Invokes a ternary function accepting single-precision floating-point number, a signed 32-bit integer, and a single-precision floating-point number and returning a single-precision floating-point number.
```
```suggestion
* Invokes a ternary function accepting a single-precision floating-point number, a signed 32-bit integer, and a single-precision floating-point number and returning a single-precision floating-point number.
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/napi/ternary/src/fif_f.c:
```diff
@@ -0,0 +1,98 @@
+		return NULL;
+	}
+
+	double x;
```
```suggestion
	float x;
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/napi/ternary/include/stdlib/math/base/napi/ternary/fif_f.h:
```diff
@@ -0,0 +1,84 @@
+			env,                                                               \
+			"exports",                                                         \
+			NAPI_AUTO_LENGTH,                                                  \
+			stdlib_math_base_napi_did_d_wrapper,                               \
```
```suggestion
			stdlib_math_base_napi_fif_f_wrapper,                               \
```

anandkaranubc committed:
chore: apply suggestions from review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc reviewed (dismissed):
Looks good! Thanks @Neerajpathak07!

kgryte commented on lib/node_modules/@stdlib/math/base/napi/ternary/README.md:
```diff
@@ -449,6 +449,69 @@ The function accepts the following arguments:
 
+#### STDLIB_MATH_BASE_NAPI_MODULE_FIF_F( fcn )
+
+Macro for registering a Node-API module exporting an interface for invoking a ternary function accepting a single-precision floating-point number, a signed 32-bit integer, a single-precision floating-point number and returning a single-precision floating-point number.
```
```suggestion
Macro for registering a Node-API module exporting an interface for invoking a ternary function accepting a single-precision floating-point number, a signed 32-bit integer, and a single-precision floating-point number and returning a single-precision floating-point number.
```

kgryte committed:
docs: update description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/math/base/napi/ternary/src/fif_f.c:
```diff
@@ -0,0 +1,98 @@
+	assert( status == napi_ok );
+
+	napi_value v;
+	status = napi_create_double( env, (double)fcn( (float)x, (int32_t)y, (float)z ), &v );
```
```suggestion
	status = napi_create_double( env, (double)fcn( (float)x, y, (float)z ), &v );
```

`y` already is `int32_t`. No need to cast again here.

kgryte committed:
refactor: remove unnecessary cast

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM
